ush/python/pygfs/task/aero_analysis.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python3
+
+import os
+import glob
+import gzip
+import tarfile
+from netCDF4 import Dataset
+from logging import getLogger
+from typing import Dict, List, Any
+
+from pygw.attrdict import AttrDict
+from pygw.file_utils import FileHandler
+from pygw.timetools import to_isotime, to_fv3time, to_timedelta
+from pygw.fsutils import rm_p
+from pygw.template import Template, TemplateConstants
+from pygw.yaml_file import YAMLFile
+from pygw.logger import logit
+from pygfs.task.analysis import Analysis
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class AerosolAnalysis(Analysis):
+    """
+    Class for global aerosol analysis tasks
+    """
+    @logit(logger, name="AerosolAnalysis")
+    def __init__(self, config):
+        super().__init__(config)
+
+        _res = int(self.config['CASE'][1:])
+        _res_enkf = int(self.config['CASE_ENKF'][1:])
+        _window_begin = self.current_cycle - to_timedelta(f"{self.config['assim_freq']}H") / 2
+
+        # Create a local dictionary that is repeatedly used across this class
+        local_dict = AttrDict(
+            {
+                'npx_ges': _res + 1,
+                'npy_ges': _res + 1,
+                'npz_ges': self.config.LEVS - 1,
+                'npz': self.config.LEVS - 1,
+                'npx_anl': _res_enkf + 1,
+                'npy_anl': _res_enkf + 1,
+                'npz_anl': self.config['LEVS'] - 1
```

Comment:
Yep, found that from debug testing, there are other things that don't work when testing after merging in @aerorahul 's suggestions. I won't be able to finish it today I don't think, so this will be part of the next commit.

Reply:
haha. obviously.